src/relentless/model/potential/dihedral.py

```diff
@@ -0,0 +1,482 @@
+
+    """
+
+    def derivative(self, type_, var, phi):
+        r"""Evaluate derivative with respect to a variable.
+
+        The derivative is evaluated using the :meth:`_derivative` function for all
+        :math:`u_{0,\lambda}(phi)`.
+
+        The derivative will be carried out with respect to ``var`` for all
+        :class:`~relentless.variable.Variable` parameters. The appropriate chain
+        rules are handled automatically. If the potential does not depend on
+        ``var``, the derivative will be zero by definition.
+
+        Parameters
+        ----------
+        _type : tuple[str]
+            The type for which to calculate the derivative.
+        var : :class:`~relentless.variable.Variable`
+            The variable with respect to which the derivative is calculated.
+        phi : float or list
+            The bond distance(s) at which to evaluate the derivative.
+
+        Returns
+        -------
+        float or numpy.ndarray
+            The bond derivative evaluated at ``phi``. The return type is consistent
+            with ``phi``.
+
+        Raises
+        ------
+        ValueError
+            If any value in ``phi`` is negative.
+        TypeError
+            If the parameter with respect to which to take the derivative
+            is not a :class:`~relentless.variable.Variable`.
+
+        """
+
+        return super().derivative(type_=type_, var=var, x=phi)
+
+    pass
+
+
+class OPLSDihedral(DihedralPotential):
+    r"""OPLS dihedral potential.
+
+    .. math::
+
+        u(\phi) = \frac{1}{2} \left( k_1 (1+\cos \phi) + k_2 (1+\cos 2\phi)
+        + k_3 (1+ \cos 3\phi) + k_4 (1+ \cos 4\phi) \right)
+
+    where :math:`\phi` is the dihedral between four bonded particles. The parameters
+    for each type are:
+
+    +-------------+--------------------------------------------------+
+    | Parameter   | Description                                      |
+    +=============+==================================================+
+    | ``k_1``     | First fitting coefficient :math:`k_1`.           |
+    +-------------+--------------------------------------------------+
+    | ``k_2``     | Second fitting coefficient :math:`k_2`.          |
+    +-------------+--------------------------------------------------+
+    | ``k_3``     | Third fitting coefficient :math:`k_3`.           |
+    +-------------+--------------------------------------------------+
+    | ``k_4``     | Fourth fitting coefficient :math:`k_4`.          |
+    +-------------+--------------------------------------------------+
+
+    Parameters
+    ----------
+    types : tuple[str]
+        Types.
+    name : str
+        Unique name of the potential. Defaults to ``__u[id]``, where ``id`` is the
+        unique integer ID of the potential.
+
+    Attributes
+    ----------
+    coeff : :class:`DihedralParameters`
+        Parameters of the potential for each type.
+
+    Examples
+    --------
+    OPLS Dihedral::
```

Comment:
Is this a specific type of dihedral? It might help someone to state which one it represents and include the reference to the paper somewhere in this description.
```suggestion
    OPLS dihedral::
```